app/models/post.rb

```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  after_create :increment_no_of_posts
+
+  def recent_5_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  private
+
+  def increment_no_of_posts
+    author.update(posts_counter: author.posts_counter + 1)
+  end
```

Comment:
- [ ] Your method uses `update` to manually increment the `posts_counter` attribute by fetching the current value, adding 1 to it, and then updating the database.
- It involves an additional database query to retrieve the current value of `posts_counter` before updating it. This is less efficient than using `increment!.`

- [ ] The `increment!` method is designed to safely update a counter or attribute in the database. It automatically generates an efficient SQL query to increment the specified column without the need to fetch the current value first. This makes it the preferred choice for incrementing counters.


### Sample code
```
  # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

Reply:
Okay!